src/RecyclableMemoryStream.cs

```diff
@@ -514,26 +523,34 @@ public override Task CopyToAsync(Stream destination, int bufferSize, Cancellatio
             }
             else
             {
-                var task = Task.Run(async () =>
+                if (this.largeBuffer == null)
                 {
-                    if (this.largeBuffer == null)
+                    if (this.blocks.Count == 1)
                     {
-                        var bytesRemaining = this.length;
-                        int currentBlock = 0;
-                        while (bytesRemaining > 0)
-                        {
-                            int amountToCopy = Math.Min(this.blocks[currentBlock].Length, bytesRemaining);
-                            await destination.WriteAsync(this.blocks[currentBlock], 0, amountToCopy, cancellationToken);
-                            bytesRemaining -= amountToCopy;
-                            ++currentBlock;
-                        }
+                        return destination.WriteAsync(this.blocks[0], 0, this.length, cancellationToken);
                     }
                     else
                     {
-                        await destination.WriteAsync(this.largeBuffer, 0, this.length, cancellationToken);
+                        return CopyToAsyncImpl();
+
+                        async Task CopyToAsyncImpl()
```

Comment:
I agree, this is a cleaner, better implementation.

Reply:
Reverted!

You should add an [.editorconfig](https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options) file to the solution to prevent these unvoluntary changes.